File: Army(java)/Warriors/Wizard.java
package Warriors;

import EnumID.*;
import Basic.*;

public class Wizard extends SpellCaster {
    private WarriorID warrior;
    private MagicianID magician;

    public Wizard(String name,  int hp, int mana, int dmg) {
        super(name, hp, mana, dmg);
        this.warrior = WarriorID.WIZARD;
        this.magician = MagicianID.SORCERER;
    }

    public WarriorID getWarriorID() {
        return warrior;
    }

    public void setWarriorID(WarriorID newID) {
        warrior = newID;
    }

    public MagicianID getMagicianID() {
        return magician;
    }
}

File: Army(c++)/Warriors/Wizard.cpp
#include "Wizard.hpp"
#include "../State/State.hpp"

Wizard::Wizard(const std::string& name,  int hp, int mana, int dmg) 
    : SpellCaster(name, hp, mana, dmg, new SoldierState(this))
    , warrior(WarriorID::WIZARD), magician(MagicianID::SORCERER) {}

Wizard::~Wizard() {}

WarriorID Wizard::getWarriorID() const {
    return this->warrior;
}

void Wizard::setWarriorID(WarriorID newID) {
    this->warrior = newID;
}

MagicianID Wizard::getMagicianID() const {
    return this->magician;
}
